sencha-workspace/SlateDemonstrationsStudent/app/controller/MainController.js

```diff
@@ -4,8 +4,8 @@
  *
  * TODO - Replace this content of this view to suite the needs of your application.
  */
-Ext.define('SlateDemonstrationsStudent.view.main.MainController', {
-    extend: 'Ext.app.ViewController',
+Ext.define('SlateDemonstrationsStudent.controller.MainController', {
+    extend: 'Ext.app.Controller',
```

Comment:
@aliyeysides I find it hard to believe this was the only change necessary

the `alias` isn't needed for non-viewcontroller controllers, remove it.

Also these event handlers won't work like the did as a viewcontroller. They worked with the viewcontroller because the view has `listeners` configured with these handlers identified by string name. You need to convert those into `control` configs on the global controller